### PR TITLE
Post-mock PE UG updates

### LINKED ISSUE #213: [PE-D][Tester B] Archive and Unarchive not listed in the user guide features section
while they are features, the archive and unarchive functions are not listed in the feature section of the user guide

-------------
Labels: `type.DocumentationBug` `severity.Low`
original: ChinYanXu/ped#6

### LINKED ISSUE #214: [PE-D][Tester B] Uses of the archive and delete functions
It is unclear what is the difference between archive and delete since both can be used to remove students from the list and their actions can be undone with unachive and restore respectively

-------------
Labels: `type.FeatureFlaw` `severity.Low`
original: ChinYanXu/ped#7

### LINKED ISSUE #215: [PE-D][Tester B] How to use the archive and unarchive function is not stated in the usage section of the user guide
It is not explained in the user guide usage section on how to use the archive and unarchive functions even though both functions are features of the program.

-------------
Labels: `type.DocumentationBug` `severity.Medium`
original: ChinYanXu/ped#8

### LINKED ISSUE #220: Re-implement edit function
- [x] Make archive list static 
- [x] Make edit closer to one-shot command
- [x] Better user prompts
- [x] Edit input-parsing for uppercase/lowercase

### PR DESCRIPTION
**Pull request by blackmirag3**
closes #215 
closes #214 
closes #213
closes #220

### CODE REVIEW

File: docs/UserGuide.md [hunk 1]
```diff
@@ -235,6 +236,21 @@ Subject: subject
 Current marks out of 100: 100.0
 Classes Attended: 1
 ```
+For any student that exists within the programme's master, archive and recently deleted list, the user cannot add new students with a name-phone number pair that matches that of the existing student.
+
+The following is an example of trying to add a new student that matches the name and phone number of an already existing student.
+
```
Comment (ParthGandhiNUS) [inline docs/UserGuide.md:242]: Perhaps can change to another case which doesn't show the identity of one of our code testers

File: docs/UserGuide.md [hunk 2]
```diff
@@ -275,7 +306,7 @@ Name of student to edit (blank to exit):
 Student details: 
 Name: Vincent
 
-Phone Number: 99999999
+Phone Number: 83227277
```
Comment (ParthGandhiNUS) [inline docs/UserGuide.md:309]: Keep previous phone number please

File: docs/UserGuide.md [hunk 3]
```diff
@@ -299,6 +331,67 @@ Do you want to add another subject and grade? (yes/no)
...
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 ```
+If no student is found in the master student list, edit mode will not be entered and the programme will resume as per normal.
+
+
+#### No student found to edit:
+```
+edit
+Name of student to edit (blank to exit):
+tim lau kar meeng
```
Comment (ParthGandhiNUS) [inline docs/UserGuide.md:341]: Once again we dont wanna use a name which might be associated with someone. Perhaps can stick with the Mario theme

File: docs/UserGuide.md [hunk 4]
```diff
@@ -373,6 +468,22 @@ List of students with phone numbers:
...
+Choose the type of list to display: (index only)
+1. Full student list
+2. List of students with total classes attended
+3. List of students with phone number shown
+4. List of students in archive
+5. List of students in recently deleted
+Enter your choice (1, 2, 3, 4 or 5):
+4
+List of students in archive: 
+1.Timothy
```
Comment (ParthGandhiNUS) [inline docs/UserGuide.md:480]: Same as above

File: docs/UserGuide.md [hunk 5]
```diff
@@ -299,6 +331,67 @@ Do you want to add another subject and grade? (yes/no)
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+```
+
+### Undo a delete and restore the deleted student to the working list `undo`
+Undoes the latest "delete student". Used by typing `undo`
+
+Format: `undo`
+
+```
+Example:
```
Comment (alalal47) [inline docs/UserGuide.md:388]: should standardise if we are including the word example in the example itself.

Comment (alalal47) [reply docs/UserGuide.md:388]: i believe its just a minor oversight and you had planned to remove it

**Review by alalal47 — CHANGES_REQUESTED** [review]

**Pull request merged**